src/DataStructures/Python/DataVector.cpp

```diff
@@ -11,32 +10,26 @@
 #include "DataStructures/DataVector.hpp"
 #include "PythonBindings/BoundChecks.hpp"
 #include "Utilities/GetOutput.hpp"
+
 namespace bp = boost::python;
 
-// 3 "make" functions for 3 different constructors
+namespace {
 std::shared_ptr<DataVector> make_datavector_from_list(bp::list l) {
   DataVector t = DataVector(bp::len(l));
   for (int i = 0; i < bp::len(l); i++) {
     t[i] = bp::extract<double>(l[i]);
   }
   return std::make_shared<DataVector>(std::move(t));
 }
-std::shared_ptr<DataVector> make_datavector_from_size(int size) {
-  return std::make_shared<DataVector>(static_cast<size_t>(size));
-}
-std::shared_ptr<DataVector> make_datavector_from_size_value(int size,
-                                                            double val) {
-  return std::make_shared<DataVector>(static_cast<size_t>(size), val);
-}
-
+}  // namespace
 namespace py_bindings {
 void bind_datavector() {
   // Wrapper for basic DataVector operations
-  bp::class_<DataVector>("DataVector", bp::no_init)
+  bp::class_<DataVector>("DataVector", bp::init<size_t>())
       // Each "make" function gets its own constructor
       .def("__init__", bp::make_constructor(&make_datavector_from_list))
-      .def("__init__", bp::make_constructor(&make_datavector_from_size))
-      .def("__init__", bp::make_constructor(&make_datavector_from_size_value))
+      .def(bp::init<size_t, double>())
+      //.def("__init__", bp::make_constructor(&make_datavector_from_array))
```

Comment:
delete line (we can defer adding this constructor, you can open an issue with a short description instead so that we remember this later)